compiler/rustc_typeck/src/check/expr.rs

```diff
@@ -995,7 +995,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         } else {
             // Defer other checks until we're done type checking.
             let mut deferred_cast_checks = self.deferred_cast_checks.borrow_mut();
-            match cast::CastCheck::new(self, e, t_expr, t_cast, t.span, expr.span) {
+            match cast::CastCheck::new(self, e, t_expr, e.span, t_cast, t.span, expr.span) {
```

Comment:
You should be able to remove the new `Span` parameter, and just use `self.expr.span` inside of `CastCheck`

Reply:
You are right; it seems that I got confused with `e.span` and `expr.span`. Thanks for pointing this out, I have fixed it now.